### PR TITLE
darwin: fix generate-sdk-packages.sh

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

The script is broken, since `https://opensource.apple.com/text/macos-1101.txt` is no longer available.

Also, #201348: the packages seem to have different sha256 now.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/darwin/apple-source-releases/generate-sdk-packages.sh [hunk 1]
```diff
@@ -16,14 +16,14 @@ outfile="$sdkName.nix"
 {"
 
 parse_line() {
-    readarray -t -d$'\t' package <<<$2
+    readarray -t -d$'-' package < <(printf "%s" $2)
     local pname=${package[0]} version=${package[1]}
 
     if [ -d $pname ]; then
         sha256=$(nix-prefetch-url "https://opensource.apple.com/tarballs/$pname/$pname-$version.tar.gz")
```
Comment (toonn) [inline pkgs/os-specific/darwin/apple-source-releases/generate-sdk-packages.sh:23]:
```suggestion
        sha256=$(nix-prefetch-url "https://github.com/apple-oss-distributions/$pname/archive/refs/tags/$pname-$version.tar.gz")
```

**Review by toonn — CHANGES_REQUESTED** [review]
LGTM
Let's change the tarball URL too though, since it's a redirect now anyway.

File: pkgs/os-specific/darwin/apple-source-releases/generate-sdk-packages.sh (hunk 1 above)
Comment (wegank) [reply pkgs/os-specific/darwin/apple-source-releases/generate-sdk-packages.sh]: Done. Thanks!

**Review by toonn — APPROVED** [review]

**Comment by toonn** [comment]
One nice addition would be printing usage information instead of erroring when run without arguments. I'm OK with merging this as-is though.

**Comment by toonn** [comment]
@wegank, would that be a simple addition?

**Comment by wegank** [comment]
@toonn Sorry, I'm not good at writing docs. Only added a dummy one.

**Pull request merged**